src/main/java/tr/com/orioninc/laborant/controller/RestAdminController.java

```diff
@@ -0,0 +1,57 @@
+package tr.com.orioninc.laborant.controller;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import tr.com.orioninc.laborant.model.Lab;
+import tr.com.orioninc.laborant.service.AdminService;
+
+import java.util.List;
+
+@RequestMapping("/v1")
+@AllArgsConstructor
+@RestController
+@Log4j2
+public class RestAdminController {
+    private AdminService adminService;
+
+    @GetMapping("/labs")
+    public ResponseEntity<List<Lab>> getAllLabs() {
+        return ResponseEntity.ok(adminService.getAllLabs());
+    }
+    @GetMapping("/labs/{labName}")
+    public ResponseEntity<Lab> getLab(@PathVariable("labName") String labName) {
+        return ResponseEntity.ok(adminService.getLab(labName));
+    }
+    @PostMapping("/labs/{labName}")
+    public ResponseEntity<Lab> addNewLab(@PathVariable String labName, @RequestBody Lab lab) {
+        lab.setLabName(labName);
+        Lab result = adminService.addNewLab(lab);
+        if (result == null) {
+            return ResponseEntity.badRequest().build();
+        }
+        else {
+            return ResponseEntity.ok(result);
+        }
+    }
+
+    @DeleteMapping("/labs/{labName}")
+    public ResponseEntity<String> deleteLab(@PathVariable("labName") String labName) {
+        if (adminService.deleteLabByName(labName)) {
+            log.info("Lab deleted");
+            return ResponseEntity.ok("Lab deleted");
+        }
+        else {
+            log.info("Lab not found");
+            return ResponseEntity.badRequest().body("Lab not found");
```

Comment:
its not a bad request lab is not found.
use notFound()

Reply:
The HyperText Transfer Protocol (HTTP) 400 Bad Request response status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (for example, malformed request syntax, invalid request message framing, or deceptive request routing).
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400